### PR TITLE
Error in PosConsensusFeature.InitializeAsync #2846

### PR DESCRIPTION
**Pull request by MithrilMan**
about #2846 
this fixes the exception message itself (triggers the guard) but the reason why this exception was rised ins't clear yet.
This can be merged as it is, the problem probably should be fixed on the sidechain repo

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/PosConsensusFeature.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ public override Task InitializeAsync()
             NetworkPeerConnectionParameters connectionParameters = this.connectionManager.Parameters;
 
             // Replace CMB.
-            bool oldCMBRemoved = connectionParameters.TemplateBehaviors.Remove(connectionParameters.TemplateBehaviors.Single(x => x is ConsensusManagerBehavior));
+            bool oldCMBRemoved = connectionParameters.TemplateBehaviors.Remove(connectionParameters.TemplateBehaviors.FirstOrDefault(x => x is ConsensusManagerBehavior));
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Consensus/PosConsensusFeature.cs]: I think we should maybe stop the node if we don't have a `ConsensusManagerBehavior` as we should always have one. In the BaseFeature only, 5 of then are initialized and you can't run a node without the base feature.

So maybe check oldCMBRemoved is null and then throw an exception.

Comment (MithrilMan) [reply src/Stratis.Bitcoin.Features.Consensus/PosConsensusFeature.cs]: actually it does the same, if null it will throw on the guard

**Comment by fassadlr** [comment]
@MithrilMan I actually saw this error yesterday as well. But couldnt reproduce it. I t might be worth me spending a couple of hours to see if I can get the error again. I have no idea why `connectionParameters` will not have all the behaviors by that time.

**Comment by MithrilMan** [comment]
@fassadlr that's weird, as long as we initialize features in the same order it should work.
Not sure if this can be an async issue

**Pull request merged**